### DISCUSSION TITLE
setup() does not complete on sample code

### DISCUSSION THREAD

**Discussion by mottibz** [pycaret/pycaret#2788] [Q&A]
Hello,
I am using VScode and Python 3.8.10. I created a clean setup, created a virtual environment, installed Pycaret successfully.
I then tried to run a sample:
```
from pycaret.datasets import get_data
data = get_data('diabetes')
from pycaret.classification import setup
s = setup(data, target = 'Class variable')

```
The setup() function runs forever and never gets to the point that I can click Return to continue. 
It keeps the status at "Processing Data" and never completes.
How can I resolve that? Thanks a lot!

**Comment by chm85** [comment]
Have you tried creating a fresh python environment? I am guessing there are some conflicts with other libraries. 

- build env: python -m venv demo_pycaret
- start env: Scripts\activate.bat
- pip install pycaret

**Comment by mottibz** [comment]
Hi chm85, Yes, this is a clean install and what you wrote is what I did.
1. Is there a way to get setup() to print out additional status information while processing (more than just "Processing Data")? This may help trace the issue.
2. Any additional ideas what to check?
Thanks a lot!

**Comment by brunocalcada** [comment]
Hi there,

Do you pass by the option to continue if you agree with the categorical/numerical features correct?

Perhaps try to add the code to extract the best models, as such:
best_model = compare_models()
scoring_best_model = pull()
print(scoring_best_model)

**Reply by mottibz** [reply]
Hi, I do not get to the point in which I can type "Enter" to Continue... It is stuck on "Processing Data"...
- What is the option you mentioned to continue without typing "Enter"?
- Any help will be highly appreciated!
Thanks to all!

**Comment by jagadeesanm-jag** [comment]
Do we have any update on this issue

**Reply by mottibz** [reply]
No update, no solution :-(

**Reply by jagadeesanm-jag** [reply]
@chm85  any solution on this?  Due to this blocker , my evaluation will be blocked
